File: Python/Secuenciadores/Errores.py
# TypeError
# El TypeError se produce cuando se intenta realizar una operación con un tipo de datos no válido. Por ejemplo:

# Suma de un entero y una cadena de caracteres
x = 10
y = "5"
z = x + y  # Genera un TypeError

# El mensaje de error generado sería algo así como:
# TypeError: unsupported operand type(s) for +: 'int' and 'str'




# IndexError
# El IndexError se produce cuando se intenta acceder a un índice que está fuera del rango de una secuencia. Por ejemplo:

# Acceso a un elemento inexistente de una lista
lista = [1, 2, 3]
elemento = lista[3]  # Genera un IndexError

# El mensaje de error generado sería algo así como:
# IndexError: list index out of range




# KeyError
# El KeyError se produce cuando se intenta acceder a una clave que no existe en un diccionario. Por ejemplo:
# Acceso a una clave inexistente en un diccionario
diccionario = {"nombre": "Juan", "edad": 25}
valor = diccionario["direccion"]  # Genera un KeyError

# El mensaje de error generado sería algo así como:
KeyError: 'direccion'






# ValueError
# El ValueError se produce cuando se intenta realizar una operación con un valor no válido. Por ejemplo:
# Conversión de una cadena de caracteres no numérica a entero
numero = int("abc")  # Genera un ValueError


# El mensaje de error generado sería algo así como:
# ValueError: invalid literal for int() with base 10: 'abc'





# SyntaxError
# El SyntaxError se produce cuando hay un error de sintaxis en el código. Por ejemplo:
# Falta de paréntesis en una llamada de función
resultado = print("Hola, mundo!"  # Genera un SyntaxError



# El mensaje de error generado sería algo así como:
# SyntaxError: unexpected EOF while parsing

File: Go/Secuenciadores/Errores/Errores.go
package main

import "fmt"

func main() {
	// Definir un slice de enteros
	numeros := []int{1, 2, 3, 4, 5}

	//SyntaxError
	//Este se produce cuando hay errores en el código. En este caso, faltaría el paréntesis final de la llamada a la variable
	cadena_string :="Ejemplo de error"
	fmt.Println(cadena_string

	// Iterar sobre los números
	for i := 0; i <= len(numeros); i++ {
		fmt.Println(numeros[i])
	}
	//El mensaje de error generado sería:
	//panic: runtime error: index out of range [5] with length 5
	/*Esto se debe a que la condición debería de ser i < len(numeros), ya que el índice
	del slice da inicio en el 0 y la longitud de esta es de 5, por lo tanto, el índice máximo válido es 4
	Asimismo, si se trata de imprimir el último número, es decir, el 5, dará error de índice fuera de rango
	*/
}
